server/upgrade.js

```diff
@@ -0,0 +1,70 @@
+const mongo   = require("promised-mongo")
+const db      = mongo("mongodb://localhost/bydesign", ["authors", "posts", "metadata", "categories"])
+const console = require("beautiful-log")
+const config  = require("../config")
+const utils   = require("./utils")
+
+let isLessThan = utils.isLessThan
+
+/* 
+ * upgradeToOne: Version 1 adds categories to the database
+ *     It defaults to setting all current articles to
+ *     "Technology" or whichever label is first in the
+ *     config file
+ */
+let upgradeToOne = (version) => {
+	console.warn(`Upgrading database from version ${version} to 1`)
+	let promises = []
+
+	// Add in a categories table with each of the config's
+	// categories in it
+	for (let category of config.categories) {
+		promises.push(db.categories.insert({
+			category: category.name,
+			shortname: category.shortname,
+		}))
+	}
+
+	// Update all the posts with the default category
+	// Then return the current version number
+	return db.posts.find({})
+	  .then((posts) => {
+		let promises = []
```

Comment:
I'd avoid reusing the name if possible.  Maybe `categoryPromises` and `postPromises`?